src/interfaces/permissions/request/method.ts

```diff
@@ -0,0 +1,40 @@
+import type { InterfaceMethod } from '../../types';
+import type { JsonPermissionsRequest } from './types';
+
+import { MessageResult } from '../../../response';
+import { PermissionsRequest } from './message';
+
+export const processPermissionsRequest: InterfaceMethod = async (
+  ctx,
+  message,
+  messageStore,
+  didResolver
+): Promise<MessageResult> => {
+  const request = new PermissionsRequest(message as JsonPermissionsRequest);
+
+  if (ctx.tenant !== request.grantedBy && ctx.tenant !== request.grantedTo) {
+    return new MessageResult({
+      status: { code: 400, message: 'grantedBy or grantedTo must be the targeted message recipient' }
+    });
+  }
+
+  // TODO: should we add an explicit check to ensure that there's only 1 signer?
```

Comment:
Yes, should. Actually, I didn't think we have a scenario for multi-sig right? I thought we switched to general JWS just to be consistent with general JWE.

Reply:
done: https://github.com/TBD54566975/dwn-sdk-js/commit/049a4ec93daa55bacf64cf3e3adb6610710740f2